### PR TITLE
Deletes get_kubernetes_nodes from integration tests

### PR DESCRIPTION
**Pull request by dposada**
## Changes proposed in this PR

Removing functions and tests that rely on fetching nodes directly from Kubernetes.

## Why are we making these changes?

- It introduces flakiness in the test suite
- It makes the test less black-box
- In many cases, it's ill-conceived because it's not taking into account, for example, the mappings from compute clusters to Cook pools

See also comments in the PR providing rationale for deleting at the individual test level.

### CODE REVIEW

File: integration/tests/cook/test_basic.py [hunk 1]
```diff
@@ -417,46 +417,6 @@ def test_workdir_volume_overlap(self):
         finally:
             util.kill_jobs(self.cook_url, [job_uuid], assert_response=False)
 
-    def test_no_cook_executor_on_subsequent_instances(self):
```
Comment (dposada) [inline integration/tests/cook/test_basic.py:420]: Relies on `util.node_count()` and is specific to Mesos

File: integration/tests/cook/test_basic.py [hunk 2]
```diff
@@ -1413,7 +1373,6 @@ def test_change_failed_retries(self):
         finally:
             util.kill_jobs(self.cook_url, jobs)
 
-    @unittest.skipIf(util.has_one_agent(), 'Test requires multiple agents')
```
Comment (dposada) [inline integration/tests/cook/test_basic.py:1416]: This check is not needed -- it's acceptable to assume that tests must be run in environments with more than one node

File: integration/tests/cook/test_basic.py [hunk 3]
```diff
@@ -1538,40 +1497,6 @@ def test_error_while_creating_job(self):
         self.assertEqual(resp.status_code, 400)
         self.assertEqual(f'Duplicate job uuids: ["{job1["uuid"]}"]', resp.json()['error'], resp.text)
 
-    @unittest.skipIf(util.has_ephemeral_hosts(), util.EPHEMERAL_HOSTS_SKIP_REASON)
-    @pytest.mark.xfail
-    def test_hostname_equals_job_constraint(self):
```
Comment (dposada) [inline integration/tests/cook/test_basic.py:1543]: Relies on `util.hostnames_to_consider`, is specific to non-autoscaling environments, and is already marked `xfail`

File: integration/tests/cook/test_basic.py [hunk 4]
```diff
@@ -2150,42 +2073,6 @@ def get_docker_info():
             util.session.delete('%s/rawscheduler?job=%s' % (self.cook_url, job_uuid))
             mesos.dump_sandbox_files(util.session, instance, job)
 
-    def test_unscheduled_jobs(self):
```
Comment (dposada) [inline integration/tests/cook/test_basic.py:2153]: Relies on `util.max_cpus`, is not smart (requires creating 100 max-cpu job submissions), and there are other tests that test the `/unscheduled_jobs` endpoint

File: integration/tests/cook/test_basic.py [hunk 5]
```diff
@@ -2199,223 +2086,6 @@ def test_unscheduled_jobs_partial(self):
         self.assertEqual(1, len(jobs))
         self.assertEqual(job_uuid_1, jobs[0]['uuid'])
 
-    @pytest.mark.xfail
-    def test_unique_host_constraint(self):
```
Comment (dposada) [inline integration/tests/cook/test_basic.py:2203]: Relies on `util.num_hosts_to_consider` and is already marked `xfail`

File: integration/tests/cook/test_basic.py [hunk 6]
```diff
@@ -2199,223 +2086,6 @@ def test_unscheduled_jobs_partial(self):
...
-                    self.logger.info(f'Job is now running: {job}')
-                    self.assertEqual('running', job['status'])
-                    self.assertNotIn(job['instances'][0]['hostname'], hosts)
-                else:
-                    self.fail(f'Expected job to either not be possible to place or to be running: {job}')
-        finally:
-            util.kill_jobs(self.cook_url, uuids)
-
-    @pytest.mark.xfail
-    def test_balanced_host_constraint_cannot_place(self):
```
Comment (dposada) [inline integration/tests/cook/test_basic.py:2271]: Relies on `util.num_hosts_to_consider` and is already marked `xfail`

File: integration/tests/cook/test_basic.py [hunk 7]
```diff
@@ -2199,223 +2086,6 @@ def test_unscheduled_jobs_partial(self):
...
-                self.assertNotIn(job['instances'][0]['hostname'], hosts)
-            else:
-                self.fail(f'Expected job to either not be possible to place or to be running: {job}')
-        finally:
-            util.kill_jobs(self.cook_url, uuids)
-
-    @unittest.skipIf(util.using_kubernetes(), "Kubernetes pod deletion is slow enough that the resources consumed by "
-                                              "pods in the process of being deleted interfere with this test.")
-    # Test passes fine in isolation. We should revisit this when we come up with our kubernetes integration test story.
-    def test_balanced_host_constraint_can_place(self):
```
Comment (dposada) [inline integration/tests/cook/test_basic.py:2340]: Relies on `util.num_hosts_to_consider` and is Mesos specific

File: integration/tests/cook/test_basic.py [hunk 8]
```diff
@@ -2199,223 +2086,6 @@ def test_unscheduled_jobs_partial(self):
...
-            jobs = util.wait_for_jobs_in_statuses(self.cook_url, uuids, ['running', 'completed'])
-            hosts = [j['instances'][0]['hostname'] for j in jobs]
-            host_count = Counter(hosts)
-            self.assertGreaterEqual(len(host_count), minimum_hosts, hosts)
-            self.assertLessEqual(max(host_count.values()), max_jobs_per_host, host_count)
-        finally:
-            util.kill_jobs(self.cook_url, uuids)
-
-    @unittest.skipIf(util.has_ephemeral_hosts(), util.EPHEMERAL_HOSTS_SKIP_REASON)
-    def test_attribute_equals_hostname_constraint(self):
```
Comment (dposada) [inline integration/tests/cook/test_basic.py:2368]: Relies on `util.max_node_cpus` and is specific to non-autoscaling environments

File: integration/tests/cook/test_basic.py [hunk 9]
```diff
@@ -2974,93 +2641,6 @@ def test_reconciliation(self):
             self.assertEqual(1, len(instances))
             self.assertEqual('Mesos task reconciliation', instances[0]['reason_string'])
 
-    @unittest.skipUnless(util.using_data_local_fitness_calculator(), "Requires the data local fitness calculator")
-    def test_data_local_constraint_missing_data(self):
```
Comment (dposada) [inline integration/tests/cook/test_basic.py:2978]: Data local is an unfunded, unused feature

File: integration/tests/cook/test_basic.py [hunk 10]
```diff
@@ -3212,35 +2792,6 @@ def test_command_length_limit(self):
         finally:
             util.kill_jobs(self.cook_url, [job_uuid], assert_response=False)
 
-    # Marked xfail due to flaky build: https://travis-ci.org/twosigma/Cook/jobs/594639302
-    @unittest.skipUnless(util.supports_exit_code() and not util.has_one_agent(),
-                         "Requires exit code support and multiple agents")
-    @pytest.mark.xfail
-    def test_cook_instance_num(self):
```
Comment (dposada) [inline integration/tests/cook/test_basic.py:3219]: Relies on `util.has_one_agent` and is already marked `xfail`

File: integration/tests/cook/test_pools.py [hunk 11]
```diff
@@ -1,89 +0,0 @@
...
-        util.init_cook_session(cls.cook_url)
-
-    def setUp(self):
-        self.cook_url = type(self).cook_url
-        self.logger = logging.getLogger(__name__)
-        self.user_factory = util.UserFactory(self)
-
-    @unittest.skipUnless(util.are_pools_enabled(), 'Pools are not enabled on the cluster')
-    @pytest.mark.xfail
-    def test_pool_scheduling(self):
```
Comment (dposada) [inline integration/tests/cook/test_pools.py:29]: Relies on `util.node_pool` and is already marked `xfail`

**Review by nsinkov — APPROVED** [review]

**Pull request merged**